mantidproject/mantid
# Add window setting (Floating/ontop) to workbench

**Description of work.**

This PR adds a setting to workbench to allow the floating and ontop behaviour of new windows to be toggled. Currently, this behaviour is limited to new plots, but could be extended to other windows, such as interfaces.

After a lot of trial-and-error, it was realised that a compromise would have to be reached with the settings. Ideally, we would get the same behaviour as on Windows, where ontop means:

- The window stays ontop of the main workbench window
- We have the common window functionality of minimize and maximize etc..

However, on Mac and Linux we cannot achieve the second goal. Hence, we have to compromise, this has been explained in the new documentation describing this setting. If the user wants a minimize button, they will have to change the setting to floating and possibly use one of the native desktop windowing options (e.g right click on the window on Linux) to control the priority of the window.

**To test:**
On any/every OS we expect the following behavior:

- Ontop: The window will always stay above the workbench mainwindow
- Floating: The window can appear above or below depending on whether to focus is.

1. Code Review
2. Go to settings and change the new option to ontop
3. Open a plot and check that the plot stays ontop
4. Change to floating and check that the window is indeed floating


Fixes #28306  
Refs #29736






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
